### PR TITLE
docs: update L7 traffic CiliumClusterwideEnvoyConfig example

### LINKED ISSUE #27408: CiliumClusterwideEnvoyConfig deprecated for Envoy load balancing and URL rewriting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

according to the log from cilium-agent and https://github.com/envoyproxy/envoy/issues/21464
```
level=info msg="regenerating all endpoints" reason="Envoy Listeners added or deleted" subsys=endpoint-manager
level=info msg="[cds: add 2 cluster(s), remove 8 cluster(s)" subsys=envoy-upstream threadID=50
level=info msg="[cds: added/updated 0 cluster(s), skipped 2 unmodified cluster(s)" subsys=envoy-upstream threadID=50
level=info msg="[lds: remove listener 'default/cilium-ingress-default-basic-ingress/listener'" subsys=envoy-upstream threadID=50
level=info msg="regenerating all endpoints" reason="Envoy Listeners deleted" subsys=endpoint-manager
level=warning msg="[gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) /envoy-lb-listener/envoy-lb-listener: Didn't find a registered implementation for 'envoy.filters.http.router' with type URL: ''" subsys=envoy-config threadID=50
level=warning msg="NACK received for versions after 19 and up to 20; waiting for a version update before sending again" subsys=xds xdsAckedVersion=19 xdsClientNode="host~127.0.0.1~no-id~localdomain" xdsDetail="Error adding/updating listener(s) /envoy-lb-listener/envoy-lb-listener: Didn't find a registered implementation for 'envoy.filters.http.router' with type URL: ''\n" xdsNonce=20 xdsStreamID=2 xdsTypeURL=type.googleapis.com/envoy.config.listener.v3.Listener
level=warning msg="Failed to update CiliumClusterwideEnvoyConfig" ciliumClusterwideEnvoyConfigName=envoy-lb-listener error="NACK received: Error adding/updating listener(s) /envoy-lb-listener/envoy-lb-listener: Didn't find a registered implementation for 'envoy.filters.http.router' with type URL: ''\n" k8sApiVersion= k8sUID=fe5aecd7-702f-4f68-a8f7-278b6a7baf3e subsys=k8s-watcher
```
the document of https://github.com/envoyproxy/envoy/issues/21464 needed to update.

### Cilium Version

root@instance-00qqerhq:~/cilium-mesh# cilium version
cilium-cli: v0.14.8 compiled with go1.20.4 on linux/amd64
cilium image (default): v1.13.3
cilium image (stable): v1.14.0
cilium image (running): v1.14.0

### Kernel Version

Linux instance-00qqerhq 5.8.0-050800-generic #202008022230 SMP Sun Aug 2 22:33:21 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

root@instance-00qqerhq:~/cilium-mesh# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:20:07Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.17", GitCommit:"a7736eaf34d823d7652415337ac0ad06db9167fc", GitTreeState:"clean", BuildDate:"2023-06-15T02:30:30Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.27) and server (1.22) exceeds the supported minor version skew of +/-1

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tanjunchen**
Fixes: #27408 

```release-note
docs: update L7 traffic CiliumClusterwideEnvoyConfig example
```

### CODE REVIEW

**Review by sayboras — COMMENTED** [review]
Thanks for your PR, can you help to update other files in the same dir

- envoy-admin-listener.yaml
- envoy-prometheus-metrics-listener.yaml

Also, please add similar details in the commit message.

**Comment by tanjunchen** [comment]
> Thanks for your PR, can you help to update other files in the same dir
> 
> * envoy-admin-listener.yaml
> * envoy-prometheus-metrics-listener.yaml
> 
> Also, please add similar details in the commit message.

done. @sayboras

File: examples/kubernetes/servicemesh/envoy/envoy-admin-listener.yaml [hunk 1]
```diff
@@ -31,3 +31,5 @@ spec:
           skip_xff_append: true
           http_filters:
           - name: envoy.filters.http.router
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
```
Comment (sayboras) [inline examples/kubernetes/servicemesh/envoy/envoy-admin-listener.yaml]: nit: newline

**Review by sayboras — APPROVED** [review]
One small nitpick, the rest looks good to me. Thanks.

File: examples/kubernetes/servicemesh/envoy/envoy-admin-listener.yaml (hunk 1 above)
Comment (tanjunchen) [reply examples/kubernetes/servicemesh/envoy/envoy-admin-listener.yaml]: ok. thanks for your review. updated.

**Comment by tanjunchen** [comment]
> /test

@sayboras  can you merge this pr? thanks~

**Comment by sayboras** [comment]
Sorry for late reply, I was off a last few days.

**Pull request merged**